synapse/storage/databases/main/event_federation.py

```diff
@@ -1583,6 +1583,45 @@ def _record_event_failed_pull_attempt_upsert_txn(
 
         txn.execute(sql, (room_id, event_id, 1, self._clock.time_msec(), cause))
 
+    @trace
+    async def get_event_ids_with_failed_pull_attempts(
+        self, event_ids: StrCollection
+    ) -> StrCollection:
```

Comment:
TODO: Update to use `StrSequence` when https://github.com/matrix-org/synapse/pull/15615 merges

Reply:
FWIW *Returning* a concrete type is usually fine (i.e. `List[str]` here), unless you really do want to ensure callers don't know it is a list.

(Also -- the docstring here needs to be updated.)

Reply:
> (Also -- the docstring here needs to be updated.)

Which part? It looks right to me?

Are we sensitive about list vs collection? I just used "list" colloquially and will have to update to "sequence" after using `StrSequence` in that case

Reply:
Updated to use `List[str]` here :+1: